### PR TITLE
[URP] RTHandles Load/Store actions optimizations fix

### PR DESCRIPTION
**Pull request by manuele-bonanno**
---
### Purpose of this PR
The recent RTHandle adoption disabled all the store action optimizations. It also highlighted some URP tech debt, since the load actions logic was not correct.

This PR fixes the store action optimization bug and rewrites the logic that sets up the scriptable render passes load actions.

---
### Testing status
Tested locally in Metal editor, iPhone11

---

### CODE REVIEW

**Review by sandy-carter-unity — APPROVED** [review]

**Review by jonuuukas — APPROVED** [review]

File: com.unity.render-pipelines.universal/Runtime/ScriptableRenderer.cs [hunk 1]
```diff
@@ -987,18 +987,6 @@ protected void SetupRenderPasses(in RenderingData renderingData)
...
                 rendererFeatures[i].SetupRenderPasses(this, in renderingData);
             }
-
-            // Remove any null render pass that might have been added by user by mistake
-            int count = activeRenderPassQueue.Count;
-            for (int i = count - 1; i >= 0; i--)
-            {
-                if (activeRenderPassQueue[i] == null)
-                    activeRenderPassQueue.RemoveAt(i);
-            }
```
Comment (phi-lira) [inline com.unity.render-pipelines.universal/Runtime/ScriptableRenderer.cs:997]: Why do we remove this?

**Review by phi-lira — APPROVED** [review]
Missing changelogs.

File: com.unity.render-pipelines.universal/Runtime/ScriptableRenderer.cs (hunk 1 above)
Comment (manuele-bonanno) [reply com.unity.render-pipelines.universal/Runtime/ScriptableRenderer.cs:999]: this was the copy paste error in the RTHandles PR which disabled by mistake all the store actions optimizations (the same code is already in AddRenderPasses)

File: com.unity.render-pipelines.universal/Runtime/Passes/ColorGradingLutPass.cs [hunk 2]
```diff
@@ -183,8 +183,10 @@ public override void Execute(ScriptableRenderContext context, ref RenderingData
 
                 renderingData.cameraData.xr.StopSinglePass(cmd);
 
+                cmd.SetGlobalTexture(ShaderPropertyId.sourceTex, m_InternalLut);
+                CoreUtils.SetRenderTarget(cmd, m_InternalLut, RenderBufferLoadAction.DontCare, RenderBufferStoreAction.Store, ClearFlag.None, Color.clear);
                 // Render the lut
-                cmd.Blit(null, m_InternalLut.nameID, material);
+                cmd.Blit(null, BuiltinRenderTextureType.CurrentActive, material, 0);
```
Comment (phi-lira) [inline com.unity.render-pipelines.universal/Runtime/Passes/ColorGradingLutPass.cs:189]: I'm confused why we still rely on BuiltinRenderTextureType after RTHandle. Why is that?

**Review by phi-lira — APPROVED** [review]

**Pull request merged**